test/ruby/test_data.rb

```diff
@@ -215,6 +215,20 @@ def test_with
     assert_raise_with_message(ArgumentError, "wrong number of arguments (given 1, expected 0)") do
       source.with({ bar: 2 })
     end
+
+    oddclass = Data.define(:odd) do
```

Comment:
Can you please add a structuring comment like "Testing #initialize is called" or something? The rest of the method has brief structuring, and I believe it is helpful for maintainers.

Reply:
Rather I'd prefer splitting this test method per the comments later.

Reply:
Or this :shrug: 
(Honestly, I miss RSpec's nesting. It might become unhinged sometimes, but at least `describe '#initialize` with nested examples and contexts really helps with reading and running tests.)